modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/VariableAdapter.java

```diff
@@ -880,8 +882,9 @@ private void initRange() {
                 for (UnitCharReplacement replacement: UNIT_CHARS_REPLACEMENTS) {
                     unitString = replacement.replace(unitString);
                 }
-                unit = Unit.valueOf(unitString);
-            } catch (IllegalArgumentException iae) {
+                unit = SimpleUnitFormat.getInstance().parse(unitString);
```

Comment:
May need to use Units.parseUnit(unitStirng) to access configured units?

Reply:
In Units [1], we define "kelvin" as the label for Kelvin unit. You might want to change it to "K".

Replacing ``` SimpleUnitFormat.getInstance().parse(unitString);``` by ```Units.parseUnit()``` is also a good idea, since it ensures that our units are registered on SimpleUnitFormat (Geotools units are registered during static code execution of Units class).

[1] https://github.com/dispiste/geotools/blob/0b824095825e9287f7e2a16a983a1aa984d29fc2/modules/library/referencing/src/main/java/org/geotools/measure/Units.java#L136